src/Sparrow/ByteString.cs

```diff
@@ -737,10 +817,19 @@ private SegmentInformation AllocateSegment(int size)
 
         private void AllocateExternalSegment(int size)
```

Comment:
No, there is no promise of any particular size. You allocate whatever you want. The only promise is that it cannot be lower than 64Kb.